plugins/org.polarsys.capella.scenario.editor.dsl/src/org/polarsys/capella/scenario/editor/dsl/validation/TextualScenarioValidator.xtend

```diff
@@ -243,6 +266,95 @@ class TextualScenarioValidator extends AbstractTextualScenarioValidator {
 			)
 		}
 	}
+	
+	/*
+	 * Expression shall not be empty
+	 */
+	@Check
+	def checkCombinedFragmentEmptyExpression(CombinedFragment combinedFragment) {
+		if (combinedFragment.expression == null || combinedFragment.expression.isEmpty) {
+			error(
+				'Expression can not be empty',
+				TextualScenarioPackage.Literals.COMBINED_FRAGMENT__EXPRESSION
+			)
+		}
+	}
+
+	/*
+	 * Expression shall not be empty
+	 */
+	@Check
+	def checkOperandEmptyExpression(Operand operand) {
+		if (operand.expression == null || operand.expression.isEmpty) {
+			error(
+				'Expression can not be empty',
+				TextualScenarioPackage.Literals.OPERAND__EXPRESSION
+			)
+		}
+	}
+
+	/*
+	 * Else keyword shall be put on a combined fragment that is ALT
+	 */
+	@Check
+	def checkElseKeyworkAvailable(Operand operand) {
+		if (operand.eContainer instanceof CombinedFragment) {
+			var combinedFragment = operand.eContainer as CombinedFragment
+			if (combinedFragment.keyword == 'alt' && !combinedFragment.operands.isEmpty) {
+				if (operand.keyword != 'else') {
+					error(
+						'Expected \'else\' keyword',
+						TextualScenarioPackage.Literals.OPERAND__KEYWORD
+					)
+				}
+			}
+		}
+	}
+
+	/*
+	 * No keyword shall be put on a combined fragment that is not ALT
+	 */
+	@Check
+	def checkKeyworkNotAvailable(Operand operand) {
+		if (operand.eContainer instanceof CombinedFragment) {
+			var combinedFragment = operand.eContainer as CombinedFragment
+			if (combinedFragment.keyword != 'alt' && !combinedFragment.operands.isEmpty) {
+				if (operand.keyword != null || !operand.keyword.isEmpty) {
+					error(
+						'Unexpected keyword',
+						TextualScenarioPackage.Literals.OPERAND__KEYWORD
+					)
+				}
+			}
+		}
+	}
+
+	/*
+	 * No keyword shall be put on a combined fragment that is not ALT
+	 */

```

Comment:
this description doesn't match the method

Reply:
Done